### PR TITLE
TINKERPOP-1715: update spark version to 2.2

### PR DESCRIPTION
**Pull request by artem-aliev**
That required:
  - more spark conflicting dependency exclusion in pom.xml
  - more spark and scala classes registration in tinkerPop gryo

### CODE REVIEW

File: gremlin-groovy/pom.xml [hunk 1]
```diff
@@ -93,6 +93,10 @@ limitations under the License.
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (artem-aliev) [inline gremlin-groovy/pom.xml]: Hm, It was needed for build.....
Now I can not reproduce the failure, so rollback it

**Comment by spmallette** [comment]
All tests pass with `docker/build.sh -t -i`

VOTE +1

**Comment by okram** [comment]
VOTE +1

**Comment by davebshow** [comment]
VOTE +1

**Pull request merged**